nullaway/src/main/java/com/uber/nullaway/generics/GenericsChecks.java

```diff
@@ -443,6 +487,58 @@ public static void checkTypeParameterNullnessForAssignability(
     }
   }
 
+  private @Nullable Type inferMethodTypeArgument(
```

Comment:
For both this method and the next one (`replaceGenerics`), I think you need a proper visitor rather than just a recursive function.  E.g., what if you have a generic method with return type `Foo<U>[]`, or `Foo<Foo<U>[]>`?  I don't think the current code will handle it.